modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/MuleArtifactResourcesRegistry.java

```diff
@@ -282,9 +288,9 @@ public DefaultClassLoaderManager getArtifactClassLoaderManager() {
   }
 
   /**
-   * @return the {@link ToolingApplicationDescriptorFactory}.
+   * @return the {@link AbstractDeployableDescriptorFactory}.
    */
-  public ToolingApplicationDescriptorFactory getToolingApplicationDescriptorFactory() {
+  public AbstractDeployableDescriptorFactory<MuleApplicationModel, ApplicationDescriptor> getToolingApplicationDescriptorFactory() {
```

Comment:
if possible return an interface instead of an abstract class.

Reply:
the interface does not have all the methods required by the caller while this abstract class does. And this abstract class was already API before these changes, so no problem using that.